src/evict.c

```diff
@@ -411,15 +415,29 @@ int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *lev
     mem_used = (mem_used > overhead) ? mem_used - overhead : 0;
 
     /* Compute the ratio of memory usage. */
-    if (level) *level = (float)mem_used / (float)server.maxmemory;
+    if (level) {
+        if (server.maxmemory_reserved_scale)
+            *level = (float)mem_used / (float)server.maxmemory_available;
+        else
+            *level = (float)mem_used / (float)server.maxmemory;
+    }
 
-    if (mem_reported <= server.maxmemory) return C_OK;
+    if (server.maxmemory_reserved_scale) {
+        if (mem_reported <= server.maxmemory_available) return C_OK;
+    } else if (mem_reported <= server.maxmemory)
+        return C_OK;
 
     /* Check if we are still over the memory limit. */
-    if (mem_used <= server.maxmemory) return C_OK;
+    if (server.maxmemory_reserved_scale) {
+        if (mem_used <= server.maxmemory_available) return C_OK;
+    } else if (mem_used <= server.maxmemory)
+        return C_OK;
 
     /* Compute how much memory we need to free. */
-    mem_tofree = mem_used - server.maxmemory;
+    if (server.maxmemory_reserved_scale) {
+        mem_tofree = mem_used - server.maxmemory_available;
+    } else
+        mem_tofree = mem_used - server.maxmemory;
```

Comment:
is this check needed? if `server.maxmemory_reserved_scale` is 0, `server.maxmemory_available` would be equal to `server.maxmemory`.

similarly, this would apply to the other changes in this function.

Reply:
Address.